commands/registry.go

```diff
@@ -247,21 +245,20 @@ func RunRegistryLogin(c *CmdConfig) error {
 		}
 		user, pass := splitCreds[0], splitCreds[1]
 
-		// log in via the docker cli
-		args := []string{
-			"login", host,
-			"-u", user,
-			"--password-stdin",
+		authconfig := configtypes.AuthConfig{
+			Username:      user,
+			Password:      pass,
+			ServerAddress: host,
 		}
-		cmd := execCommand("docker", args...)
-		cmd.Stdin = strings.NewReader(pass)
-		cmd.Stdout = c.Out
-		cmd.Stderr = c.Out
 
-		err = cmd.Run()
+		cf := dockerconf.LoadDefaultConfigFile(os.Stderr)
+		dockerCreds := cf.GetCredentialsStore(authconfig.ServerAddress)
+		err = dockerCreds.Store(authconfig)
```

Comment:
I think these two operations call cgo. I'm going to pull down your branch right now to check, but if so, it might make our release process more complicated since we may not be able to use Go's built-in cross compilation as part of goreleaser.

Reply:
Okay – they do call cgo, but it seems like it does not affect our ability to use Go's usual cross compilation tools, so I think this is fine.